cdk/stack.py

```diff
@@ -181,21 +255,30 @@ def build_api_lambda(
             )
         )
 
-        data_pipeline_arn = env.get("DATA_PIPELINE_ARN")
-        handler.add_to_role_policy(
-            iam.PolicyStatement(
-                actions=["states:StartExecution"],
-                resources=[data_pipeline_arn],
+        if data_pipeline_arn := env.get("DATA_PIPELINE_ARN"):
+            handler.add_to_role_policy(
+                iam.PolicyStatement(
+                    actions=["states:StartExecution"],
+                    resources=[data_pipeline_arn],
+                )
             )
-        )
-        handler.add_to_role_policy(
-            iam.PolicyStatement(
-                actions=["states:DescribeExecution", "states:GetExecutionHistory"],
-                resources=[
-                    f"{env.get('DATA_PIPELINE_ARN').replace(':stateMachine:', ':execution:')}*"  # noqa
-                ],
+            handler.add_to_role_policy(
+                iam.PolicyStatement(
+                    actions=["states:DescribeExecution", "states:GetExecutionHistory"],
+                    resources=[
+                        f"{env.get('DATA_PIPELINE_ARN').replace(':stateMachine:', ':execution:')}*"  # noqa
+                    ],
+                )
             )
-        )
+            handler.add_to_role_policy(
+                iam.PolicyStatement(
+                    actions=["airflow:CreateCliToken"],
+                    resources=[
+                        f"arn:aws:airflow:us-west-2:853558080719:environment/{env.get('MWAA_ENV')}"
```

Comment:
style nit: maybe we just need an MWAA_AIRFLOW_ARN in  our config which defaults false? Elsewhere we could use `if config.mwaa_airflow_arn:` do aiflow deployment things? Would fix the linting error anyway 🙃

Reply:
Oh I see that you do use the mwaa_env string when creating a token so the just use an ARN in config suggestion may not be a universal fit on it's own.